### PR TITLE
[VYMCA-95] UI import issue fix for migrate_source_csv v3

### PR DESCRIPTION
**Pull request by hamrant**
## Steps for review:

Jira - https://fivejars.atlassian.net/browse/VYMCA-95

- [ ]  Login as admin
- [ ]  Go to `/admin/openy/virtual-ymca/gc-auth-settings`
- [ ]  Enable **`Custom provider`** and disable `Dummy provider`
- [ ] Edit **`Custom provider`**
- [ ] Check that you can see new section with migration info

screenshot-carnation openy docksal-2020 10 16-13_14_33

- [ ] Press on the first link
- [ ] Create CSV file

```csv
Member ID,Member First Name,Primary Member,Member Email,Package Name,Package Site
309302,Test,Yes,TEST@GMAIL.COM,OLY - Family - Staff,Oscar Lasko Branch
309303,Jon,No,JON@GMAIL.COM,OLY - Family - Staff,Oscar Lasko Branch
117892,Doe,Yes,Doe@GMAIL.COM,West Chester - SilverSneakers - S & PP,West Chester
91896,Mike,No,SUPERMIKE@GMAIL.COM,Kennett - Family - Full,Kennett Branch
```

- [ ] Upload file in the form and submit

Screenshot from 2020-10-16 13-15-48

- [ ] Return to settings form with migration info and open the second link

screenshot-carnation openy docksal-2020 10 16-13_20_31

- [ ] Press `Execute`
- [ ] Go to `/admin/people`
- [ ] And check that users from your file imported to site
- [ ] Profit

### CODE REVIEW

**Comment by hamrant** [comment]
retest

File: modules/openy_gc_auth/modules/openy_gc_auth_custom/src/Form/ImportCsvForm.php [hunk 1]
```diff
@@ -108,10 +113,37 @@ public function buildForm(array $form, FormStateInterface $form_state) {
...
+    $form_state->setRedirectUrl($url);
+
+    $fid = $form_state->getValue(['csv_file', 0]);
+    $csv_file_id = $form_state->get('csv_file_id');
+    if ($fid == $csv_file_id) {
+      return;
+    }
+
+    $fid = $form_state->getValue(['csv_file', 0]);
+    $csv_file_id = $form_state->get('csv_file_id');
```
Comment (NightHunterSV) [inline modules/openy_gc_auth/modules/openy_gc_auth_custom/src/Form/ImportCsvForm.php:134]: The same assignments are done a few lines above.

File: modules/openy_gc_auth/modules/openy_gc_auth_custom/openy_gc_auth_custom.info.yml [hunk 2]
```diff
@@ -7,8 +7,8 @@ core: 8.x
 core_version_requirement: ^8 || ^9
 dependencies:
   - drupal:migrate
-  - drupal:migrate_source_csv
-  - drupal:migrate_plus
+  - drupal:migrate_source_csv (>=8.x-3.4)
+  - drupal:migrate_plus (>=8.x-5.1)
```
Comment (NightHunterSV) [inline modules/openy_gc_auth/modules/openy_gc_auth_custom/openy_gc_auth_custom.info.yml:11]: Please, add also the `migrate_tools` dependency

File: modules/openy_gc_auth/modules/openy_gc_auth_custom/src/Form/ImportCsvForm.php (hunk 1 above)
Comment (hamrant) [reply modules/openy_gc_auth/modules/openy_gc_auth_custom/src/Form/ImportCsvForm.php]: fixed

File: modules/openy_gc_auth/modules/openy_gc_auth_custom/openy_gc_auth_custom.info.yml (hunk 2 above)
Comment (hamrant) [reply modules/openy_gc_auth/modules/openy_gc_auth_custom/openy_gc_auth_custom.info.yml:11]: fixed

**Review by NightHunterSV — APPROVED** [review]

**Review by anpolimus — APPROVED** [review]

**Comment by anpolimus** [comment]
code is ok

**Comment by anpolimus** [comment]
code tested

**Comment by abrochka** [comment]
retest

**Pull request merged**